test/extended/machines/workers.go

```diff
@@ -122,64 +107,65 @@ var _ = g.Describe("[Feature:Machines][Disruptive] Managed cluster should", func
 		skipUnlessMachineAPIOperator(c.CoreV1().Namespaces())
 
 		g.By("validating node and machine invariants")
-		// fetch machines
-		machines, err := listMachines(dc, machineLabelSelectorWorker)
+		// fetch all machines
+		allMachines, err := listMachines(dc, "")
 		if err != nil {
 			e2e.Failf("unable to fetch worker machines: %v", err)
 		}
-		numMachineWorkers := len(machines)
-		if numMachineWorkers == 0 {
-			e2e.Failf("cluster should have worker machines")
+
+		if len(allMachines) == 0 {
+			e2e.Failf("cluster should have machines")
 		}
 
 		// fetch nodes
 		nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{
 			LabelSelector: nodeLabelSelectorWorker,
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
+		if len(nodes.Items) == 0 {
+			e2e.Failf("cluster should have nodes")
+		}
 		// map node -> machine
-		nodeToMachine, nodeMatch := mapNodeNameToMachineName(nodes.Items, machines)
+		nodeToMachine, nodeMatch := mapNodeNameToMachine(nodes.Items, allMachines)
 		if !nodeMatch {
 			e2e.Failf("unable to map every node to machine.  nodeToMachine: %v, nodeName: %v", nodeToMachine, nodeNames(nodes.Items))
 		}
-		machineToNode, machineMatch := mapMachineNameToNodeName(machines, nodes.Items)
-		if !machineMatch {
-			e2e.Failf("unable to map every machine to node.  machineToNode: %v, machineNames: %v", machineToNode, machineNames(machines))
+		workerMachines := make([]objx.Map, len(nodeToMachine))
+		for _, m := range nodeToMachine {
+			workerMachines = append(workerMachines, m)
 		}
+		numMachineWorkers := len(workerMachines)
```

Comment:
`numMachineWorkers := len(machineToNode)` will do as well [1]:
```
len(s)    string type      string length in bytes
          [n]T, *[n]T      array length (== n)
          []T              slice length
          map[K]T          map length (number of defined keys)
          chan T           number of elements queued in channel buffer
```

[1] https://golang.org/ref/spec#Length_and_capacity

Reply:
sure we can do that. But i am slightly leaning towards creating and using `workerMachines` for better code readability.

Reply:
You can then rename `machineToNode` to `machineToWorkerNode`